### DISCUSSION TITLE
convert webp to tiff

### DISCUSSION THREAD

**Discussion by beiwei38233** [ImageMagick/ImageMagick#6685] [Help]
### ImageMagick version

7.1.1.17

### Operating system, version and so on

win10

### Description

I want to convert a webp photo to be a tiff by using magickwand. How should I do?

**Comment by beiwei38233** [comment]
Here's the simple code:
#include <stdio.h>
#include <MagickWand/MagickWand.h>

int main() {
    // 初始化MagickWand
    MagickWandGenesis();

    // 创建一个MagickWand对象
    MagickWand* webp = NewMagickWand();
    MagickReadImage(webp, "B255.webp");
    MagickWriteImage(webp, "1.bmp");
    
    DestroyMagickWand(webp);

    MagickWand* bmp = NewMagickWand();
    MagickReadImage(bmp, "1.bmp");
    MagickWriteImage(bmp, "1.tif");
    
    // 清理资源
    DestroyMagickWand(bmp);

    //MagickWand
    MagickWandTerminus();

    return 0;
}

It will get a worry tiff photo. But when i use: magick convert B255.webp 1.tif, then i will get a right. Can you tell me why?

**Comment by snibgo** [comment]
MagickReadImage() and MagickWriteImage() both return a status. If that is false, you should report the error.

**Reply by beiwei38233** [reply]
How do I check for this error? I use:

if(MagickReadImage(webp, "B255.webp")==MagickTrue){

printf("Read success.\n")}

Like this, and it prints success.

Can you leave your email or private chat with me? I want to send the example to you to try. Thank you
